coderd/externalauth.go

```diff
@@ -201,6 +203,24 @@ func (api *API) externalAuthCallback(externalAuthConfig *externalauth.Config) ht
 			apiKey = httpmw.APIKey(r)
 		)
 
+		extra := pqtype.NullRawMessage{}
+		if len(externalAuthConfig.ExtraTokenKeys) > 0 {
+			extraMap := map[string]interface{}{}
+			for _, key := range externalAuthConfig.ExtraTokenKeys {
+				extraMap[key] = state.Token.Extra(key)
+			}
+			extraData, err := json.Marshal(extraMap)
+			if err != nil {
+				httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+					Message: "Failed to marshal extra token keys.",
+					Detail:  err.Error(),
+				})
+				return
+			}
+			extra.RawMessage = extraData
+			extra.Valid = true
+		}
```

Comment:
Only comment is the default could just be `{}`.

Reply:
I'd rather it be null so we know if the user specified any extra.